service/src/test/java/bio/terra/profile/service/profile/flight/create/CreateProfileFlightTest.java

```diff
@@ -179,25 +144,37 @@ void createAzureProfileSuccess() {
                     .managedResourceGroupId(mrgId)));
     when(azureService.getRegisteredProviderNamespacesForSubscription(any(), any()))
         .thenReturn(azureConfiguration.getRequiredProviders());
-
-    var profile =
-        new BillingProfile(
-            UUID.randomUUID(),
-            "fake-bp-name",
-            "fake-description",
-            "direct",
-            CloudPlatform.AZURE,
-            Optional.empty(),
-            Optional.of(tenantId),
-            Optional.of(subId),
-            Optional.of(mrgId),
-            null,
-            null,
-            null);
+    var profile = ProfileFixtures.createAzureBillingProfile(tenantId, subId, mrgId);
 
     profileService.createProfile(profile, userRequest);
 
     verify(applicationService)
         .addTagToMrg(tenantId, subId, mrgId, "terra.billingProfileId", profile.id().toString());
+    verify(samService).createManagedResourceGroup(profile, userRequest);
+  }
+
+  @Test
+  void createAzureProfile_removeSamMrgOnFailure() throws InterruptedException {
+    var subId = UUID.randomUUID();
+    var tenantId = UUID.randomUUID();
+    var mrgId = "fake-mrg";
+    when(azureService.getAuthorizedManagedAppDeployments(any(), any(), any()))
+        .thenReturn(
+            Collections.singletonList(
+                new AzureManagedAppModel()
+                    .tenantId(tenantId)
+                    .subscriptionId(subId)
+                    .managedResourceGroupId(mrgId)));
+    when(azureService.getRegisteredProviderNamespacesForSubscription(any(), any()))
+        .thenReturn(azureConfiguration.getRequiredProviders());
+    doThrow(SamExceptionFactory.create("foo", new InterruptedException()))
+        .when(samService)
+        .createManagedResourceGroup(any(), eq(userRequest));
+    var profile = ProfileFixtures.createAzureBillingProfile(tenantId, subId, mrgId);
+
+    assertThrows(
+        SamInterruptedException.class, () -> profileService.createProfile(profile, userRequest));
+
+    verify(samService).deleteManagedResourceGroup(profile.id(), userRequest);
```

Comment:
This is interesting-- `createManagedResourceGroup` is what threw the exception, so I wouldn't necessarily expect `deleteManagedResourceGroup` to succeed (I'm imagining the resource group would not be have been created). But I suppose the NOT_FOUND handling catches that case.